### ISSUE TITLE
Unable to create lookup list custom fields

### ISSUE THREAD

**Issue by eavonius** [mautic/mautic#13265]
### Mautic Version

5.0.x series

### PHP version

8.1.2

### What browsers are you seeing the problem on?

Chrome

### What happened?

Trying to create a custom field, I get the usual error page in the UI. Log entries further below.

### How can we reproduce this issue?

Step 1: Go to the Custom Fields settings page.
Step 2: Click the "New" button.
Step 3: Enter the following values in the form:
   * Label: "Employment Status"
   * Alias: "employment_status"
   * Select "Professional" from the Group field
   * Select "Lookup" from the Data Type field.
   * Under Options, add 3 options:
      * Label "Employee" value "Employee" (I tried the number 1 here too)
      * Label "Self-Employed" value "Self-Employed" (I tried the number 2 here too)
      * Label "Unemployed" value "Unemployed" (I tried the number 3 here too)

Step 4: Click the "Save" or "Save and Close" buttons. You'll see the error page.

### Relevant log output

```shell
mautic.WARNING: PHP Warning - Undefined array key "list" - in file /var/www/mautic/docroot/app/bundles/LeadBundle/Form/Type/FieldType.php - at line 242 [] {"hostname":"ip-<redacted>","pid":42129}
[2024-01-23T23:45:40.663814+00:00] mautic.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "An exception has been thrown during the rendering of a template ("Field "__name__" has already been rendered, save the result of previous render call to a variable and output that instead.")." at /var/www/mautic/docroot/app/bundles/CoreBundle/Resources/views/FormTheme/mautic_form_layout.html.twig line 381 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): An exception has been thrown during the rendering of a template (\"Field \"__name__\" has already been rendered, save the result of previous render call to a variable and output that instead.\"). at /var/www/mautic/docroot/app/bundles/CoreBundle/Resources/views/FormTheme/mautic_form_layout.html.twig:381)\n[previous exception] [object] (Symfony\\Component\\Form\\Exception\\BadMethodCallException(code: 0): Field \"__name__\" has already been rendered, save the result of previous render call to a variable and output that instead. at /var/www/mautic/vendor/symfony/form/FormRenderer.php:136)"} {"hostname":"ip-<redacted>","pid":42129}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by fillon** [comment]
@eavonius I have the same error while trying to create a `gender` custom fields, data type `Lookup`.

Did you find a solution??

Mautic 5.1.0
PHP 8.1.29

**Comment by abhisekmazumdar** [comment]
Yes, I confirm that any custom field with lookup type is broken. I'm working on a fix.

**Comment by abhisekmazumdar** [comment]
The lookup field has started accepting Label | value as options, which is mostly breaking this type.

**Comment by abhisekmazumdar** [comment]
I feel this was a similar issue to this: https://github.com/mautic/mautic/issues/12208

**Comment by RCheesley** [comment]
Thanks for looking into a fix for this @abhisekmazumdar - let us know when it's ready to review!

**Comment by abhisekmazumdar** [comment]
I have created a fix and I feel it's ready for review.
https://github.com/mautic/mautic/pull/13946

**Comment by abhisekmazumdar** [comment]
I feel we can close this issue maybe ??

**Comment by RCheesley** [comment]
Thanks @abhisekmazumdar I'll close given it's been resolved by #13946.
